frontend_app_file/src/component/Kanban.jsx

```diff
@@ -0,0 +1,386 @@
+      loading: false,
+      mustSave: false,
+      board: {
+        columns: [],
+        ...JSON.parse(props.content.raw_content || '{}')
+      },
+      isDrafAvailable: !!getLocalStorageItem(
+        'rawContent',
+        props.content,
+        'kanban'
+      )
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    if (this.props.content.current_revision_id !== prevProps.content.current_revision_id) {
+      // TODO handle the "Refresh?" button
+      this.updateBoard({
+        ...this.state.board,
+        ...JSON.parse(this.props.content.raw_content || '{}')
+      }, true)
+    }
+  }
+
+  handleRestoreDraft = () => {
+    this.updateBoard(
+      JSON.parse(getLocalStorageItem('rawContent', this.props.content, 'kanban')),
+      true
+    )
+    this.setState({ isDrafAvailable: false, mustSave: true })
+  }
+
+  handleDiscardChanges = () => {
+    if (confirm(this.props.t('Are you sure you want to discard your changes?'))) {
+      this.updateBoard(JSON.parse(this.props.content.raw_content || '{}'), true)
+      removeLocalStorageItem('rawContent', this.props.content, 'kanban')
+      this.setState({ mustSave: false, isDraftAvailable: false })
+    }
+  }
+
+  handleEditCardTitle = (card) => {
+    const title = prompt(this.props.t('Please enter the new title of this card'), card.title || '')
+    if (title) {
+      this.updateCard(card, { ...card, title })
+    }
+  }
+
+  handleEditCardContent = (card) => {
+    const description = prompt(this.props.t('Please enter the new content of this card'), card.description || '')
+    if (description) {
+      this.updateCard(card, { ...card, description })
+    }
+  }
+
+  updateCard (oldCard, updatedCard) {
+    this.updateColumns(
+      this.state.board.columns.map(
+        col => ({
+          ...col,
+          cards: col.cards.map(
+            card => (
+              oldCard === card
+                ? updatedCard
+                : card
+            )
+          )
+        })
+      )
+    )
+  }
+
+  removeCard = (card) => {
+    this.updateColumns(
+      this.state.board.columns.map(
+        col => ({
+          ...col,
+          cards: col.cards.filter(c => c !== card)
+        })
+      )
+    )
+  }
+
+  renderCard = (card, arg2, arg3) => {
+    // console.log("renderCard", card, arg2, arg3)
+    return (
+      <div className='file__contentpage__statewrapper__kanban__card'>
+        <div className='file__contentpage__statewrapper__kanban__card__title'>
+          <strong onClick={() => this.handleEditCardTitle(card)}>{card.title}</strong>
+          <IconButton
+            text=''
+            icon='trash'
+            onClick={() => this.removeCard(card)}
+          />
+        </div>
+        <div
+          className='file__contentpage__statewrapper__kanban__card__description'
+          onClick={() => this.handleEditCardContent(card)}
+        >
+          {card.description}
+        </div>
+      </div>
+    )
+  }
+
+  handleAddCardClicked (column) {
```

Comment:
I think the naming usually are in the active voice in Tracim
So I would advice `handleClickAddCard` instead

Reply:
Mhm, I like your name better but I took inspiration from something else in Tracim here. I'll rename it.